src/Aspire.Hosting/ApplicationModel/ProcessedResourceExecutionConfiguration.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents the configuration (arguments and environment variables) to apply to a specific resource.
+/// </summary>
+internal class ProcessedResourceExecutionConfiguration : IProcessedResourceExecutionConfiguration
+{
+    /// <inheritdoc/>
+    public required IEnumerable<object> References { get; init; }
+
+    /// <inheritdoc/>
+    public required IEnumerable<(object Unprocessed, string Processed, bool IsSensitive)> ArgumentsWithUnprocessed { get; init; }
+
+    /// <inheritdoc/>
+    public IEnumerable<(string Value, bool IsSensitive)> Arguments => ArgumentsWithUnprocessed.Select(arg => (arg.Processed, arg.IsSensitive));
```

Comment:
If the implementation for processed vs unprocessed Arguments and EnvironmentVariables is so similar, maybe we just have one variant (with everything, e.g. the `EnvironmentVariables` returns `IEnumerable<(name, value, unprocessed)>` ) and let the user ignore/consume the unprocessed part as necessary...

Reply:
I tried that approach, but only one or two places actually care about the unprocessed values and it pushes extra processing code onto the places that don't.

Reply:
sounds good